# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSink.h
# Commit: 3000680ad5f4
# Full Hash: 3000680ad5f46c43b409bdd5468dff476634aec7
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-04-20 21:53:00
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Wait for the audio callbacks to start being called before using the audio clock. r=alwu,media-playback-reviewers
#   
#   Because we're generally using high latency on the cubeb stream used by
#   AudioStream instance, it can take some time for the callbacks to start being
#   called, and the for the audio clock (cubeb_stream_get_position(...)) to advance.
# ==============================================================================

diff -r 048e93d0a274 -r 3000680ad5f4 dom/media/mediasink/AudioSink.h
--- a/dom/media/mediasink/AudioSink.h	Wed Apr 20 12:07:09 2022 +0000
+++ b/dom/media/mediasink/AudioSink.h	Wed Apr 20 12:07:10 2022 +0000
@@ -77,6 +77,12 @@
 
   const RefPtr<AudioDeviceInfo>& AudioDevice() { return mAudioDevice; }
 
+  // This returns true if the audio callbacks are being called, and so the
+  // audio stream-based clock is moving forward.
+  bool AudioStreamCallbackStarted() {
+    return mAudioStream && mAudioStream->CallbackStarted();
+  }
+
  private:
   // Allocate and initialize mAudioStream. Returns NS_OK on success.
   nsresult InitializeAudioStream(const PlaybackParams& aParams);